go-controller/pkg/ovn/pods.go

```diff
@@ -374,30 +376,44 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			return fmt.Errorf("unable to create LSPSetDynamicAddresses command for port: %s", portName)
 		}
 		cmds = append(cmds, cmd)
-	} else {
+
+		// ensure we have/can allocate the IP in the annotation
+		if err = oc.lsManager.AllocateIPs(logicalSwitch, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
+			// this should also be treated as an allocation for purposes of error handling
+			releaseIPs = true
```

Comment:
@trozet per below, if we fail to reserve the already-annotated IPs, we can't just reallocate them. Gotta set needsIP=false here and do something intelligent with the error.